machine-runner/README.md

```diff
@@ -232,6 +232,70 @@ Instead, each robot independently ensures that after at five seconds a decision
+
+To avoid errors, the general best practice is to not stash commands in an external variable and defer the call.
+The `commands()` will return undefined when possible errors are detected.
+The passing of time might have invalidated the detection result, which is the stashed value.
+
+```typescript
+// Good
+await whenParticularState.commands()?.someCommand()
+
+// Bad
+const commands = whenParticularState.commands();
+await someLongRunningTask();   // this may have invalidated the error detection when finished executing 
+await commands?.someCommand(); 
+```
+
+The list of errors that may arise is as follows.
+
+- `MachineRunnerErrorCommandFiredAfterExpired`:
+
+  This error results from a command call to an expired state.
+
+  An "Expired" state object is a state object that does not match its machine-runner's current state object. An expired state object can be obtained by means of storing a state object's reference in a variable and using it at a later time when the machine-runner has transitioned to another state.
+
+- `MachineRunnerErrorCommandFiredAfterLocked`:
+
+  This error results from a command call to a locked state object/machine-runner. It is generally avoidable by not having a command being called twice in a concurrent fashion in the same state.
+
+  "Locked" is a transitionary state object/machine-runner between the time a command is called and it receives a rejection. When a command results in a failed publication, the source state objects are unlocked, thus subsequent commands will be available. When a command results in a successful publication, the machine-runner is unlocked, but the origin state object is kept locked to prevent subsequent command calls.
+
+- `MachineRunnerErrorCommandFiredAfterDestroyed`:
+
+  This error results from a command call to a state object of a destroyed machine.
+
+  "Destroyed" is the status of a machine-runner that has been destroyed, either by explicitly calling its `.destroy()` method or by breaking out of the `for-await` that is applied to the runner.
+
+- `MachineRunnerErrorCommandFiredWhenQueueNotEmpty`:
+
+  This error results from a command call to a state object when the machine is in the middle of processing multi-events transitions.
+
+  Machine-runner supports multi-event transitions. In a multi-events transition, a runner buffers events of an incomplete ordered events chain in a queue. If the queue is not empty, or in other words, the machine-runner is waiting for a later part of an events chain, then it will not be able to issue a command.
+
+- `MachineRunnerErrorCommandFiredWhenNotCaughtUp`:
+
+  This error results from a command call to a state object whose machine is not caught up.
+
+  "Caught up" is a state where a machine-runner has processed all published events in a subscription stream. Actyx sends events in batches. During the processing of a batch, the machine-runner is not caught up.
```

Comment:
perhaps note that this error can be avoided by only invoking commands on the results of `next()` and `peek()`, which includes for-await loops

Reply:
This should be caught by the `command()` getter. Although I will add this in the best practice section above, worded a bit differently.